macros/contexts/contextNondecimalBase.pl

```diff
@@ -0,0 +1,268 @@
+
+=head1 NAME
+
+contextNonDecimalBase.pl - Implements a MathObject class and context for numbers
+in non-decimal bases
+
+=head1 DESCRIPTION
+
+This context implements positive integers and some operations on integers in a non-decimal base
+greater than or equal to 2.  The numbers will be stored internally in decimal, though parsed
+and shown in the chosen base.
+
+In addition, basic integer arithemetic (+,-,*,^) are available for these numbers, however because it
+is assumed that all numbers are integers, division '/' is disallowed.
```

Comment:
It might be better to implement integer division (result is essentially `int(A/B)`) rather than remove it entirely.

Reply:
I was looking a `contextFraction.pl` as an example and thought that I would need to make 
````package context::NondecimalBase::BOP::divide;```
and define division to be `int(A/B)`.  And then (similar to `contextFraction.pl`), defined a `_eval` subroutine.  

I added: 
```
$context->operators->set(
		'/'  => { class => 'context::NondecimalBase::BOP::divide'},
		' /' => { class => 'context::NondecimalBase::BOP::divide'},
		'/ ' => { class => 'context::NondecimalBase::BOP::divide'}
	);
```
to the `new` sub starting on line 152.  I'm not getting the `_eval` called above.  

Is this the right way to go and I'm missing something or is there an easier way?

Reply:
It's hard to tell without seeing the actual implementation of the `context::NondecimalBase::BOP::divide`.  Can you post the package here?

Also, you may need to override the `div` method in `context::NondecimalBase::Real` to do the integer division.  The original is

https://github.com/openwebwork/pg/blob/a29b85f259fac1924688a2a6f8b80b9715c4bc55/lib/Value/Real.pm#L89-L93

so you probably only need to add `int()` around the result of the division in the last line of the function.